strawberry/experimental/pydantic/object_type.py

```diff
@@ -43,17 +51,36 @@
 from strawberry.types.type_resolver import _get_fields
 from strawberry.types.types import TypeDefinition
 
-from .exceptions import MissingFieldsListError, UnregisteredTypeException
+from .exceptions import MissingFieldsListError
+
+
+try:
+    from typing import ForwardRef  # type: ignore
+except ImportError:  # pragma: no cover
+    # ForwardRef is private in python 3.6 and 3.7
+    from typing import _ForwardRef as ForwardRef  # type: ignore
+
+from .lazy_types import LazyForwardRefType, LazyModelType
 
 
-def replace_pydantic_types(type_: Any, is_input: bool):
+def replace_pydantic_types(
+    type_: Any, is_input: bool, model: Type[BaseModel], name: str
+):
```

Comment:
I did also run in the Enum issue throwing \`TypeError: StrawberryModel fields cannot be resolved. \_enum\_definition\` raised in #1598. The solution suggested there, with the addition of a type check before the issubclass check, solved the issue for me, along the lines of  
  
\`\`\`python  
import inspect  
from enum import Enum  
\# ...  
    if  inspect.isclass(type\_)  and issubclass(type\_, Enum):  
       strawberry.enum(type\_)  
\`\`\`